File: src/geek time/c#/week01/24. Swap Nodes in Pairs.cs
using common;
using System;
using System.Collections.Generic;
using System.Text;

namespace week01 {
    public partial class Solution {
        public ListNode SwapPairs(ListNode head) {
            // 边界判断
            if (head == null || head.next == null) {
                return head;
            }

            var root = new ListNode();
            var pre = new ListNode();
            root.next = pre;
            var cur = head;

            while (cur != null && cur.next != null) {
                var next = cur.next.next;
                pre.next = cur.next;
                cur.next.next = cur;
                cur.next = next;

                pre = cur;
                cur = next;
            }
            return root.next.next;
        }
    }
}

File: src/geek time/python/week01/24. Swap Nodes in Pairs.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
    
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if head is None or head.next is None:
            return head
        dummy = ListNode()
        pre = ListNode()
        dummy.next = pre
        cur = head
        while cur and cur.next:
            next = cur.next.next
            pre.next = cur.next
            cur.next.next = cur
            cur.next = next
            pre = cur
            cur = next
        return dummy.next.next


